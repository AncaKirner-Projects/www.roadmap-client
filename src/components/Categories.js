import React, { Component } from 'react';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import styled from 'styled-components';

const FormGroupStyle = styled(FormGroup)`
  margin-left: 20px;
`;

class Categories extends Component {
  render() {
    return (
      <FormGroupStyle>
        {this.props.categories.map((category) =>
          <FormControlLabel key={category.id}
            control={
              <Checkbox
                color="primary"
                checked={category.checked}
                onChange={e => this.props.onChange(e)}
                value={category.category_name}
              />
            }
            label={category.category_name}
          />)}
      </FormGroupStyle>
    );
  }
}

export default Categories;
