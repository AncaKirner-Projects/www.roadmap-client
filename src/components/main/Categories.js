import React, { Component } from 'react';
import PropTypes from 'prop-types';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import styled from 'styled-components';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { selectCategory, unselectCategory } from '../../store/actions/categoryActions';

const FormGroupStyle = styled(FormGroup)`
  margin-left: 20px;
`;

class Categories extends Component {
  render() {
    const { categories } = this.props;
    return (
      <FormGroupStyle>
        {categories && categories.map((category) => {
          return (<FormControlLabel key={category.id}
            control={
              <Checkbox
                color="primary"
                checked={category.checked}
                onChange={this.handleChange(category)}
                value={category.name}
              />
            }
            label={category.name}
          />);
        }
        )}
      </FormGroupStyle>
    );
  }

  handleChange = (category) => (e) => {
    // e.preventDefault();
    category.checked = e.target.checked;

    if (e.target.checked) {
      this.props.selectCategory(category);
    }
    else {
      category.checked = !e.target.checked;
      this.props.unselectCategory(category);
    }
  }
}

Categories.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    list: PropTypes.array,
    selected: PropTypes.arrayOf(PropTypes.number.isRequired)
  })),
  selectCategory: PropTypes.func.isRequired,
  unselectCategory: PropTypes.func.isRequired
};

const mapStateToProps = (state) => {
  return {
    categories: state.categories.list,
    selected: state.categories.selected
  }
}

const mapDispatchToProps = (dispatch) => ({
  ...bindActionCreators({
    selectCategory: selectCategory,
    unselectCategory: unselectCategory
  }, dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(Categories);
