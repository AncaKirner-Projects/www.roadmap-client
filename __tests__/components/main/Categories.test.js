import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Categories from '../../../src/components/main/Categories';
import 'jest-styled-components';

const middlewares = [thunk];
const mockStore = configureMockStore();
const store = mockStore({
  selectCategory: jest.fn(() => { console.log('here'); }),
  categories: {
    list: [
      {
        id: 1,
        name: 'Categ1',
        checked: true
      },
      {
        id: 2,
        name: 'Categ2',
        checked: false
      }
    ],
    selected: [1]
  }
});

describe('<Categories />', () => {
  it('renders a card list', () => {
    const wrapper = shallow(<Categories store={store} />);
    const component = wrapper.dive().dive();

    expect(toJson(component)).toMatchSnapshot();
  });

  // it('handles a category check', () => {
  //   const wrapper = mount(<Categories store={store} />);
  //   const onChangeFake = jest.spyOn(wrapper.instance(), 'handleChange');

  //   wrapper.find('input[type="checkbox"][value="Categ1"]')
  //     .simulate('change', { target: { value: 'Categ1' } });
  //   expect(onChangeFake).toHaveBeenCalledTimes(1);
  // });
});
