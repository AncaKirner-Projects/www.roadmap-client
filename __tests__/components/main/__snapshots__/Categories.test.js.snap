// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> renders a card list 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "Naked": [Function],
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  margin-left: 20px;
",
        ],
      },
      "displayName": "Styled(WithStyles(FormGroup))",
      "options": Object {
        "name": "MuiFormGroup",
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
>
  <WithStyles(FormControlLabel)
    control={
      <WithStyles(Checkbox)
        checked={true}
        color="primary"
        onChange={[Function]}
        value="Categ1"
      />
    }
    key="1"
    label="Categ1"
  />
  <WithStyles(FormControlLabel)
    control={
      <WithStyles(Checkbox)
        checked={false}
        color="primary"
        onChange={[Function]}
        value="Categ2"
      />
    }
    key="2"
    label="Categ2"
  />
</StyledComponent>
`;
